java/src/org/openqa/selenium/firefox/FirefoxOptions.java

```diff
@@ -372,8 +372,62 @@ public FirefoxOptions merge(Capabilities capabilities) {
     Require.nonNull("Capabilities to merge", capabilities);
     FirefoxOptions newInstance = new FirefoxOptions();
     getCapabilityNames().forEach(name -> newInstance.setCapability(name, getCapability(name)));
-    capabilities.getCapabilityNames().forEach(name -> newInstance.setCapability(name, capabilities.getCapability(name)));
     newInstance.mirror(this);
+    capabilities.getCapabilityNames().forEach(name -> {
+      if (name.equals("args") && capabilities.getCapability(name) != null) {
```

Comment:
If we use `Optional.ofNullable(capabilities.getCapability(name)).orElse(Collections.emptyList())` (for all the if conditions) we can remove the null check.

Reply:
I understand your reasoning but even if that case for some values (depending on the values, I did go over the code again to check), it will have to return a null object or a null string and that might need a check. Keeping it simple in this case should be fine.